lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -431,20 +431,25 @@ xilinx::AIE::TileOp xilinx::AIE::ObjectFifoCreateOp::getProducerTileOp() {
 
 // ObjectFifoLinkOp
 LogicalResult xilinx::AIE::ObjectFifoLinkOp::verify() {
-  ObjectFifoCreateOp fifoIn = getFifoIn().getDefiningOp<ObjectFifoCreateOp>();
-  AIEObjectFifoType fifoType = fifoIn.getType().cast<AIEObjectFifoType>();
-  MemRefType elemType = fifoType.getElementType().cast<MemRefType>();
-  int inputSize = (int)elemType.getShape()[0];
-
-  for (auto fifoOut : getFifoOuts()) {
-    ObjectFifoCreateOp fifoOutOp = fifoOut.getDefiningOp<ObjectFifoCreateOp>();
-    if (fifoIn.getConsumerTiles()[0] != fifoOutOp.getProducerTile())
-      return emitError("ObjectFifoLinkOp must have a link point, i.e., a "
-                       "shared tile between objectFifos");
+  auto sharedTile = getOptionalSharedTile();
+  if (!sharedTile)
+    return emitError("ObjectFifoLinkOp must have a link point, i.e., a "
+                      "shared tile between objectFifos");
+
+  if (sharedTile) {
+  auto sharedTileOp = (*sharedTile).getDefiningOp<TileOp>();
+  for (auto user : sharedTileOp.getOperation()->getUsers())
+    if (isa<xilinx::AIE::CoreOp>(user))
+      return user->emitOpError("currently cannot be created on AIE tile used as "
```

Comment:
This might be overly restrictive. Can the restriction be limited to not allowing the Core op to interact with the object FIFOs involved in the link for now? Otherwise this effectively disables the core, if I am understanding this correctly

Reply:
Ah yes, that's a good idea! I'll add that in instead!